clippy_lints/src/types.rs

```diff
@@ -975,6 +993,24 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for CastPass {
                     },
                 }
             }
+
+            match &cast_from.sty {
+                ty::TyFnDef(..) |
+                ty::TyFnPtr(..) => {
+                    if cast_to.is_numeric() && cast_to.sty != ty::TyUint(UintTy::Usize){
+                        span_lint_and_sugg(
+                            cx,
+                            FN_TO_NUMERIC_CAST,
+                            expr.span,
+                            &format!("casting a Fn to {} may truncate the function address value.", cast_to),
+                            "if you need address of function, use cast to `usize` instead:",
+                            format!("{} as usize", &snippet(cx, ex.span, "x"))
```

Comment:
This looks good! I think for the suggestion message `"if you need the address of the function, consider:"` would be enough, since the `usize` hint is already in the suggestion.